In [1]:
from src.modules.bear_cleaner import *
from src.modules.models import *
from sklearn.model_selection import train_test_split

# Importation des données

In [2]:
df = data_model("./data/vins.json")

# Choix des variables expliqués et explicatives 

In [3]:
X = df[["capacity", "keyword_1", "keyword_2", "keyword_3", "millesime", "cepage", "par_gouts", "service",
        "avg_temp","conservation_date", "bio", "customer_fav", "is_new", "top_100", "destock", "sulphite_free",
        "alcohol_volume", "type","country", "bubbles", "wine_note", "nb_reviews"]]
y = df["unit_price"]
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

# Modèles

In [4]:
model = model_mlp(X_train, y_train)

c:\Users\guill\AppData\Local\Programs\Python\Python311\Lib\site-packages\sklearn\neural_network\_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.
  warnings.warn(


In [5]:
model_result(model=model)

+--------+-------+-------+
| Modèle | Score |   SD  |
+--------+-------+-------+
| model  | 0.419 | 0.056 |
+--------+-------+-------+


Il faut bien mettre "entrainement__nom_du_parametre_optimisé"

In [6]:
model_param(model, 
            "imputation__strategy",
            "entrainement__hidden_layer_sizes",
            "entrainement__solver")

+----------------------------------+----------+
|            Parameter             |  Value   |
+----------------------------------+----------+
|       imputation__strategy       |  median  |
| entrainement__hidden_layer_sizes | (60, 60) |
|       entrainement__solver       |   adam   |
+----------------------------------+----------+


Remember (ne noter que si on trouve mieux) : 
Meilleurs modèles :
- MLP - 0.419 imputation__strategy  : median  | entrainement__hidden_layer_sizes | (60, 60) | entrainement__solver : adam 
- RF - 0.422 - imputation__strategy : mean | entrainement__n_estimators :  20  |  entrainement__max_depth :  9 